### PR TITLE
Add `updateMapStyle` Gradle task

### PR DESCRIPTION
**Pull request by FloEdelmann**
Automate the task that caused #4071.

CC @Helium314 @ENT8R

### CODE REVIEW

File: app/src/main/assets/map_theme/jawg/global.yaml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 global:
     api_key: false # intended to be overwritten by StreetComplete
     language: false # intended to be overwritten by StreetComplete
-    language_script: Latn # intended to be overwritten by StreetComplete
```
Comment (FloEdelmann) [inline app/src/main/assets/map_theme/jawg/global.yaml:4]: Is this change intended?

`global.language_script` is still referenced in this file in line 80 and in

https://github.com/streetcomplete/StreetComplete/blob/18189f123416d9a0b332e50afab58b857452b416/app/src/main/java/de/westnordost/streetcomplete/screens/main/map/components/SceneMapComponent.kt#L69

Comment (westnordost) [reply app/src/main/assets/map_theme/jawg/global.yaml:4]: > Is this change intended?

hm, no

File: buildSrc/src/main/java/UpdateMapStyleTask.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.net.URL
+
+open class UpdateMapStyleTask : DefaultTask() {
+
+    @get:Input var targetDir: String? = null
+
+    @TaskAction fun run() {
+        println("hello")
```
Comment (westnordost) [inline buildSrc/src/main/java/UpdateMapStyleTask.kt:14]: hello

File: buildSrc/src/main/java/UpdateMapStyleTask.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+private fun String.normalizeLineEndings() = this.replace("\r\n", "\n")
+
+private fun String.replaceOnlineWithLocalSections(): String {
+    val lines = this.split("\n").toMutableList()
+
+    val onlineStartLineIndices = lines.indices.filter { lines[it].trim().startsWith("# for online testing") }
+    val localStartLineIndices = lines.indices.filter { lines[it].trim().startsWith("# for local testing") }
+
+    val onlineRanges = onlineStartLineIndices.zip(localStartLineIndices)
+    onlineRanges.forEach { (start, end) -> assert(start < end) }
```
Comment (westnordost) [inline buildSrc/src/main/java/UpdateMapStyleTask.kt:65]: `assert` is kind of deprecated in Java. Not really (I think) but the behavior is not what one would expect. IIRC it can be configured as a JVM flag what should be the behavior of assert and by default it is a no-op in release builds. In any case, assert seems to be meant as a debugging tool rather than to enforce contracts. That's what I learnt ~10 years ago at least, feel free to disabuse me.

In any case, in Kotlin there are `check` (for state checks) and `require` (for ensuring contracts).

File: buildSrc/src/main/java/UpdateMapStyleTask.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    @get:Input var targetDir: String? = null
+
+    @TaskAction fun run() {
+        println("hello")
+
+        val targetDir = targetDir?.let { File(it) } ?: return
+        assert(targetDir.exists())
+
+        val githubDirectoryListingUrl = URL("https://api.github.com/repos/streetcomplete/streetcomplete-mapstyle/contents?ref=jawg")
```
Comment (westnordost) [inline buildSrc/src/main/java/UpdateMapStyleTask.kt:19]: pass branch name as parameter instead?

**Review by westnordost — APPROVED** [review]
I think this can be merged, thank you for that work! 

In case you tasted blood and are interested in adding more automatism, there is one thing that would be most appreciated: Generating the taginfo JSON. You probably remember that it failed before.

File: app/src/main/assets/map_theme/jawg/global.yaml (hunk 1 above)
Comment (FloEdelmann) [reply app/src/main/assets/map_theme/jawg/global.yaml:4]: So this line is always manually added in the StreetComplete repo after copying the map style from the mapstyle repo?

File: buildSrc/src/main/java/UpdateMapStyleTask.kt (hunk 2 above)
Comment (FloEdelmann) [reply buildSrc/src/main/java/UpdateMapStyleTask.kt:14]: oops

File: buildSrc/src/main/java/UpdateMapStyleTask.kt (hunk 3 above)
Comment (FloEdelmann) [reply buildSrc/src/main/java/UpdateMapStyleTask.kt:65]: Ah, I always forget that. I'll change it to `require` then.

File: app/src/main/assets/map_theme/jawg/global.yaml (hunk 1 above)
Comment (westnordost) [reply app/src/main/assets/map_theme/jawg/global.yaml:4]: I think so

**Review by westnordost — APPROVED** [review]

File: app/src/main/assets/map_theme/jawg/global.yaml (hunk 1 above)
Comment (FloEdelmann) [reply app/src/main/assets/map_theme/jawg/global.yaml]: With https://github.com/streetcomplete/streetcomplete-mapstyle/pull/129, this is now automated, too (commit [`0fdeda5`](https://github.com/streetcomplete/StreetComplete/pull/4100/commits/0fdeda5fa3aa3a2ae19a510e91ec15ff29591a67) was created by the task).
